official/utils/logs/hooks.py

```diff
@@ -73,14 +73,30 @@ def __init__(self,
     self._total_steps = 0
     self._batch_size = batch_size
     self._warm_steps = warm_steps
+    self._train_op = None
```

Comment:
I think this might be applicable for all the hooks that requires accurate value for global step tensor. So probably worthwhile to push it to run_session_hook itself.